### PR TITLE
xdp workshop

### PR DESCRIPTION
**Pull request by daniel-lavie-seekret**

### CODE REVIEW

File: workshop2/demo-server/main.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018- The Pixie Authors.
```
Comment (guyarb) [inline workshop2/demo-server/main.go]: that does not belong with pixie!
you should give me the credits

File: workshop2/setup_docker.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#! /bin/bash
+
+ROOT_DIR=$(dirname $(dirname $(realpath "${0}")))
+echo $ROOT_DIR
```
Comment (guyarb) [inline workshop2/setup_docker.sh]: is it relevant?

File: workshop2/xdp_prog.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2017 GustavoKatel
+// Licensed under the Apache License, Version 2.0 (the "License")
```
Comment (guyarb) [inline workshop2/xdp_prog.c]: can you add the source link?

did you just copy the code or did you change it?

File: workshop2/xdp_prog.c [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+#define KBUILD_MODNAME "foo"
```
Comment (guyarb) [inline workshop2/xdp_prog.c]: why is it here? what is it?

File: workshop2/xdp_runner.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+#cgo LDFLAGS: -lbcc
+#include <bcc/bcc_common.h>
+#include <bcc/libbpf.h>
+void perf_reader_free(void *ptr);
+*/
+import "C"
+
+// protocols contains minimal mappings between internet protocol
+// names and numbers for platforms that don't have a complete list of
+// protocol numbers.
```
Comment (guyarb) [inline workshop2/xdp_runner.go]: sorry, but i didn't understand

File: workshop2/xdp_runner.go [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+//
+// See https://www.iana.org/assignments/protocol-numbers
+//
+// On Unix, this map is augmented by readProtocols via lookupProtocol.
```
Comment (guyarb) [inline workshop2/xdp_runner.go]: do we run on system other than unix?

**Review by guyarb — CHANGES_REQUESTED** [review]

File: workshop2/demo-server/main.go (hunk 1 above)
Comment (daniel-lavie-seekret) [reply workshop2/demo-server/main.go]: Copied it from your workshop

File: workshop2/setup_docker.sh (hunk 2 above)
Comment (daniel-lavie-seekret) [reply workshop2/setup_docker.sh]: Removed

File: workshop2/xdp_prog.c (hunk 3 above)
Comment (daniel-lavie-seekret) [reply workshop2/xdp_prog.c]: Added a link for "based on". Now it's more different after the recent changes

File: workshop2/xdp_prog.c (hunk 4 above)
Comment (daniel-lavie-seekret) [reply workshop2/xdp_prog.c]: Removed

File: workshop2/xdp_runner.go (hunk 5 above)
Comment (daniel-lavie-seekret) [reply workshop2/xdp_runner.go]: Changed

File: workshop2/xdp_runner.go (hunk 6 above)
Comment (daniel-lavie-seekret) [reply workshop2/xdp_runner.go]: Removed

File: workshop2/xdp_runner.go [hunk 7]
```diff
@@ -15,39 +15,27 @@ import (
...
-// On Unix, this map is augmented by readProtocols via lookupProtocol.
+const (
+	numberOfArguments  = 2
+	bpfDefaultLogLevel = 1
+	bpfLogSize         = 65536
+)
 
+// protocols is a mapping between a protocol number to its string representation
 var protocols = map[uint32]string{
-	0:  "HOPOPT",
```
Comment (guyarb) [inline workshop2/xdp_runner.go]: you removed it from here but the example image does have it...

File: workshop2/xdp_runner.go [hunk 8]
```diff
@@ -58,23 +46,19 @@ func main() {
 		os.Exit(1)
 	}
 
-	xdpReturnCode := "XDP_DROP"
-	module := bcc.NewModule(string(bpfSourceCodeContent), []string{
-		"-w",
-		"-DRETURNCODE=" + xdpReturnCode,
-	})
+	module := bcc.NewModule(string(bpfSourceCodeContent), []string{})
```
Comment (guyarb) [inline workshop2/xdp_runner.go]: can it work?
```suggestion
	module := bcc.NewModule(string(bpfSourceCodeContent), nil)
```

File: workshop2/xdp_prog.c [hunk 9]
```diff
@@ -1,79 +1,64 @@
...
-{
-    struct ipv6hdr *ip6h = data + nh_off;
-    if ((void *)&ip6h[1] > data_end)
-        return 0;
-    return ip6h->nexthdr;
-}
+// Main logic starts here
 
-int xdp_dropper(struct xdp_md *ctx)
+int xdp_counter(struct xdp_md *ctx)
```
Comment (guyarb) [inline workshop2/xdp_prog.c:20]: please use brackets in the entire code

File: workshop2/demo-server/main.go (hunk 1 above)
Comment (guyarb) [reply workshop2/demo-server/main.go]: so please remove from both of them

**Review by guyarb — CHANGES_REQUESTED** [review]

File: workshop2/xdp_runner.go (hunk 7 above)
Comment (daniel-lavie-seekret) [reply workshop2/xdp_runner.go]: Nice! took a new screenshot

File: workshop2/xdp_runner.go (hunk 8 above)
Comment (daniel-lavie-seekret) [reply workshop2/xdp_runner.go]: Yup, nil slice. changed

**Review by guyarb — APPROVED** [review]

**Pull request merged**